java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -247,4 +252,47 @@ public void set_missing_value_and_returnOldValue_is_null(BaseClient client) {
         String data = client.set("another", ANOTHER_VALUE, options).get();
         assertNull(data);
     }
+
+    @SneakyThrows
+    @ParameterizedTest
+    @MethodSource("getClients")
+    public void sadd_srem_scard_smembers_existing_set(BaseClient client) {
+        String key = UUID.randomUUID().toString();
+        assertEquals(4, client.sadd(key, "member1", "member2", "member3", "member4").get());
+        assertEquals(1, client.srem(key, "member3", "nonExistingMember").get());
+
+        Set<String> expectedMembers = new HashSet<>(Arrays.asList("member1", "member2", "member4"));
+        assertEquals(expectedMembers, client.smembers(key).get());
+        assertEquals(1, client.srem(key, "member1").get());
+        assertEquals(2, client.scard(key).get());
+    }
+
+    @SneakyThrows
+    @ParameterizedTest
+    @MethodSource("getClients")
+    public void srem_scard_smembers_non_existing_key(BaseClient client) {
+        assertEquals(0, client.srem("nonExistingKey", "member").get());
+        assertEquals(0, client.scard("nonExistingKey").get());
+        assertEquals(new HashSet<String>(), client.smembers("nonExistingKey").get());
+    }
+
+    @SneakyThrows
+    @ParameterizedTest
+    @MethodSource("getClients")
+    public void sadd_srem_scard_smembers_key_with_non_set_value(BaseClient client) {
+        String key = UUID.randomUUID().toString();
+        assertEquals(OK, client.set(key, "foo").get());
+
+        RedisException e = assertThrows(RedisException.class, () -> client.sadd(key, "bar").get());
```

Comment:
```suggestion
        Exception e = assertThrows(ExecutionExection.class, () -> client.sadd(key, "bar").get());
```
then `e.cause()` would be a `RedisException`

Try finding another IT with `assertThrows` for reference